### PR TITLE
add copt solver

### PR DESCRIPTION
**Pull request by agramfort**
closes #11

### CODE REVIEW

File: solvers/copt.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        f = copt.loss.LogLoss(X, y, alpha=self.lmbd / len(X))
+
+        warnings.filterwarnings('ignore', category=RuntimeWarning)
+
+        if solver == 'pgd':
+            step_size = 1.0 / f.lipschitz
+            result = cp.minimize_proximal_gradient(
+                f.f_grad,
+                np.zeros(n_features),
+                step=lambda x: step_size,
```
Comment (fabianp) [inline solvers/copt.py:50]: I think this might prevent the backtracking line-search from kicking in

File: solvers/copt.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        elif solver == 'saga':
+            step_size = 1.0 / (3 * f.max_lipschitz)
+            result = cp.minimize_saga(
+                f.partial_deriv,
+                X,
+                y,
+                np.zeros(n_features),
+                step_size=step_size,
+                tol=0,
+                max_iter=n_iter,
```
Comment (fabianp) [inline solvers/copt.py:65]:
```suggestion
                alpha=self.lmbd / len(X),
```

Comment (fabianp) [reply solvers/copt.py:65]: My suggestion was to add the alpha kwarg, not remove the max_iter

File: solvers/copt.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        f = copt.loss.LogLoss(X, y, alpha=self.lmbd / X.shape[0])
+
+        warnings.filterwarnings('ignore', category=RuntimeWarning)
+
+        if solver == 'pgd':
+            step_size = 1.0 / f.lipschitz
+            result = cp.minimize_proximal_gradient(
+                f.f_grad,
+                np.zeros(n_features),
+                step=lambda x: step_size,
```
Comment (fabianp) [inline solvers/copt.py:57]: for PGD I would not pass a step-size so that it uses backtracking

**Comment by fabianp** [comment]
other than that I would not use a fixed step-size for PGD it looks good to me!

**Comment by fabianp** [comment]
BTW I added the max_lipschitz also on the squared loss (https://github.com/openopt/copt/commit/f21b689b19dd8370c48039686a7fdd8bbbe23dfc) but not sure what needs to change in benchopt

**Comment by agramfort** [comment]
greeeeeennnnn !!!

**Pull request merged**